apps/01PigLatin.js

```diff
@@ -4,30 +4,86 @@ var assert = require('assert');
-      }
-      if (( word.indexOf('o') > -1 && word.indexOf('o') < vowelIndex) || vowelIndex === -1) {
-         vowelIndex = word.indexOf('o');
-      }
-      if (( word.indexOf('u') > -1 && word.indexOf('u') < vowelIndex) || vowelIndex === -1) {
-         vowelIndex = word.indexOf('u');
-      }
-}
+//Concept check
+//
+//Please explain why we are using String.prototype.toLowerCase() here.
+//For more info:
+//https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/toLowerCase
+//
+//Your Answer:
+//It is used to convert a string to lowercase letters, but it does not change the orginal string.
+
+word = word.toLowerCase();
+
+//Concept Check
+//
+//What is the significance of setting vowelIndex to -1?
+//This might help:
+//https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/lastIndexOf
+//
+//Your Answer:
+//To move any consonant from the beggining of the word to the end of it (e.g. cat = atc).
+
+var vowelIndex = -1;
+
+if ( ( word.indexOf('a') > -1 && word.indexOf('a') < vowelIndex ) || vowelIndex === -1 ) {
+             vowelIndex = word.indexOf('a');
+         }
+
+if ( ( word.indexOf('e') > -1 && word.indexOf('e') < vowelIndex ) || vowelIndex === -1 ) {
+             vowelIndex = word.indexOf('e');
+         }
+
+if ( ( word.indexOf('i') > -1 && word.indexOf('i') < vowelIndex ) || vowelIndex === -1 ) {
+             vowelIndex = word.indexOf('i');
+         }
+
+if ( ( word.indexOf('o') > -1 && word.indexOf('o') < vowelIndex ) || vowelIndex === -1 ) {
+             vowelIndex = word.indexOf('o');
+         }
+
+if ( ( word.indexOf('u') > -1 && word.indexOf('u') < vowelIndex ) || vowelIndex === -1 ) {
+             vowelIndex = word.indexOf('u');
+         }
+
+if ( ( word.indexOf('y') > -1 && word.indexOf('y') < vowelIndex ) || vowelIndex === -1 ) {
+             vowelIndex = word.indexOf('y');
+         }
+
+//Concept check
+//
+//Why are we using String.prototype.slice() here? What roles does vowelIndex play?
+//This might help:
+//https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/slice
+//
+//Your Answer:
+//We are using slice here to remove the first character from the vowelIndex. If the vowelIndex is equal to zero then adds 'yay' to the new string. Otherwise, return the rest of the word adding the first part of the word with ay at the end of the string.
+
+var firstPart = word.slice(0, vowelIndex);
+var restWord = word.slice(vowelIndex, word.length);
+
+   if (vowelIndex === 0) {
+     return word + 'yay';
+   }
+   else {
+     return restWord + firstPart + 'ay';
+   }
+ }
+
 
+//Concept check
+//
+//Make sure you run this app with node in the console before anwering this question.
+//Also, read Lesson 1 in the textbook.
+//http://intermediate.austincodingacademy.com/_book/lesson_one.html
+//
+//What does prompt.get() do for us?
+//
+//Your Answer:
+//It gets us the answer of the function.
```

Comment:
prompt.get() displays some text on the terminal which "prompts" the users for values. It then calls the Piglatin function that does all the work.